README.md

```diff
@@ -1,6 +1,20 @@
 # Registry Image Resource
 
 Supports checking, fetching, and pushing of images to Docker registries.
+This resource can be used in two ways: [with `tag`
+specified](#check-with-tag-discover-new-digests-for-the-tag) and [without
+`tag`](#check-without-tag-discover-semver-tags).
+
+With `tag` specified, `check` will detect changes to the digest the tag points
+to, and `out` will always push to the specified tag. This is to be used in
+simpler cases where no real versioning exists.
+
+With `tag` omitted, `check` will instead detect tags based on semver versions
```

Comment:
Since semver will be the default, will existing resources in pipelines that don't specify a `tag` will continue to maintain the current status quo where `latest` is the default ?

Reply:
Yep. It works a bit differently, since it'll change to detecting all the semver versions, and if `latest` has the same digest as a semver tag (which is also the latest version) you'll see that tag instead, but `latest` will always take precedence if it has its own special digest or if there are no semver tags at all.

So, this should be backwards compatible, with one edge case exception: if `latest` points to the same digest as a tag `1.2.3` but there is a newer version tag, say `1.3.0`, the resource will return `1.3.0` whereas before it would have returned the `1.2.3` digest.

This is a little awkward to fix because it's a state that the registry could be in temporarily while a new version is being bumped - say `1.3.0` has just shipped but the `latest` tag hasn't bumped yet. If we have `latest` take priority, the version history would go from (1.2.3) to (1.2.3, 1.3.0, latest [1.2.3]), and then finally to (1.2.3, 1.3.0). Which seems a bit janky.

Reply:
That makes sense. The probability of the repo being in the state of the edge case is low and based on the meaning of `latest`, the returning  `1.3.0` as the last version rather than `1.2.3` makes more sense logically.